src/js/libcs/OpSound.js

```diff
@@ -1,48 +1,533 @@
+    audioCtx.close();
+    audioCtx = null;
+};
 
 evaluator.playsin$1 = function(args, modifs) {
+    if (!audioCtx) {
+        audioCtx = getAudioContext();
+    }
 
     function handleModifs() {
         var erg;
+        if (modifs.amp !== undefined) {
+            erg = evaluate(modifs.amp);
+            if (erg.ctype === 'number') {
+                amp = erg.value.real;
+            }
+        }
+        if (modifs.damp !== undefined) {
+            erg = evaluate(modifs.damp);
+            if (erg.ctype === 'number') {
+                damp = erg.value.real;
+            }
+        }
+        if (modifs.duration !== undefined) {
+            erg = evaluate(modifs.duration);
+            if (erg.ctype === 'number') {
+                duration = erg.value.real;
+            }
+        }
+        if (modifs.stop !== undefined) {
+            erg = evaluate(modifs.stop);
+            if (erg.ctype === 'number') {
+                duration = erg.value.real;
+            }
+        }
+        if (modifs.harmonics !== undefined) {
+            erg = evaluate(modifs.harmonics);
+            if (erg.ctype === 'list') {
+                harmonics = erg.value;
+            }
+        }
         if (modifs.line !== undefined) {
-
             erg = evaluate(modifs.line);
             if (erg.ctype === 'number') {
-                linenumber = Math.floor(erg.value.real);
-                if (linenumber < 0) {
-                    linenumber = 0;
+                var val = erg.value.real;
+            } else if (erg.ctype === 'string') {
+                var val = erg.value;
+            } else {
+                var val = 0;
+            }
+            if (!linesDict.hasOwnProperty(val)) {
+                line = Object.keys(linesDict).length;
+                linesDict[val] = line;
+                lines.push(0);
+            } else {
+                line = linesDict[val];
+            }
+        }
+        if (modifs.partials !== undefined) {
+            erg = evaluate(modifs.partials);
+            if (erg.ctype === 'list') {
+                partials = erg.value;
+            }
+        }
+        if (modifs.restart !== undefined) {
+            erg = evaluate(modifs.restart);
+            if (erg.ctype === 'boolean') {
+                restart = erg.value;
+            }
+        }
+        if (modifs.attack !== undefined) {
+            erg = evaluate(modifs.attack);
+            if (erg.ctype === 'number') {
+                attack = erg.value.real;
+            }
+        }
+        if (modifs.release !== undefined) {
+            erg = evaluate(modifs.release);
+            if (erg.ctype === 'number') {
+                release = erg.value.real;
+            }
+        }
+    }
+
+    var freq = evaluate(args[0]).value.real;
+    var amp = 0.5;
+    var damp = 0;
+    var duration = 1;
+    var harmonics = [CSNumber.real(1)];
+    var line = 0;
+    var partials = Array(harmonics.length).fill(CSNumber.real(1));
```

Comment:
maybe the code would be more readable and error-prone if you avoid the CSNumer data structure internally for `harmonic` and `partials`.